### PR TITLE
FEAT: Implement snip-5 for SocialPay account contract

### LINKED ISSUE #76: Implement snip-5 for SocialPay account contract
## OD Hack
This issue is part of ODHack 4.0:
* description - https://x.com/OnlyDust_com/status/1791035111992934563
* guidelines - https://onlydust.notion.site/ODHack-Common-Guidelines-Project-leads-5e21836f85ad493f9ca429770af0ad40

## Task
Implement snip-5 for SocialPay account:
* account contract is here: https://github.com/keep-starknet-strange/joyboy/blob/main/onchain/src/social/account.cairo
* it should use: https://docs.openzeppelin.com/contracts-cairo/0.10.0/introspection

## References
* https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-5.md

## Communication
https://t.me/JoyboyStarknet/949

May the Joy be with you!

### PR DESCRIPTION
**Pull request by VictoriaAde**
This PR implements snip-5 for SocialPay account contract.
I imported SRC5 from [OpenZeppelin](https://github.com/OpenZeppelin/cairo-contracts.git) and also added test case to make sure our implementation is correct.

Fixes #76

### CODE REVIEW

File: onchain/src/social/account.cairo [hunk 1]
```diff
@@ -48,8 +65,12 @@ pub mod SocialAccount {
         ) { // TODO: implement handle transfer logic
         }
     }
-}
 
+    #[external(v0)]
+    fn supports_interface_fn(self: @ContractState, interface_id: felt252) -> bool {
```
Comment (maciejka) [inline onchain/src/social/account.cairo:70]: Why `fn` at the end of a function?

File: onchain/src/social/account.cairo [hunk 2]
```diff
@@ -32,10 +46,13 @@ pub mod SocialAccount {
         public_key: u256
     }
 
+    const ISRC5_ID: felt252 = 0x123456789abcdef0;
```
[the remark this answers is not in the document]
Comment (maciejka) [reply onchain/src/social/account.cairo:49]: There should be a script to calculate id of `ISocialAccount`?

**Review by maciejka — CHANGES_REQUESTED** [review]

File: onchain/src/social/account.cairo (hunk 1 above)
Comment (VictoriaAde) [reply onchain/src/social/account.cairo]: Oh my bad, It was an oversight.
I have given it a better name.

**Comment by VictoriaAde** [comment]
@maciejka, I have been fixing some errors that came from the recent push to the main branch but I can't seem to fix this.

Any way you can help?

image

**Comment by maciejka** [comment]
> @maciejka, I have been fixing some errors that came from the recent push to the main branch but I can't seem to fix this.
> 
> Any way you can help?
> 
> image

Se if use use the same versions:
image

**Comment by VictoriaAde** [comment]
> > @maciejka, I have been fixing some errors that came from the recent push to the main branch but I can't seem to fix this.
> > Any way you can help?
> > image
> 
> Se if use use the same versions: image

I updated my snforge and it's still giving same error 
Screenshot from 2024-05-27 22-25-23

**Comment by maciejka** [comment]
try to move `deploy_erc20` helper

**Comment by maciejka** [comment]
The work is continued in an another pr: #125

**Pull request closed without merging**